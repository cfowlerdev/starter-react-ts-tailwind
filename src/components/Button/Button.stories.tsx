import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import { Button } from './Button';

export default {
  title: 'Button',
  component: Button,
  argTypes: {}
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Standard = Template.bind({});
Standard.args = {};
