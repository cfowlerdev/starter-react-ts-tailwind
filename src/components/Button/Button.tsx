import React from 'react';

interface IProps {}

export const Button: React.FC<IProps> = () => {
  return <button>Hello</button>;
};
