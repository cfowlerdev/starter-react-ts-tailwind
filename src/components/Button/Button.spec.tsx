import { render, screen } from '@testing-library/react';
import { Button } from './Button';

describe('<Button />', () => {
  it('should render in primary colour and medium size by default', () => {
    const { container } = render(<Button />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
