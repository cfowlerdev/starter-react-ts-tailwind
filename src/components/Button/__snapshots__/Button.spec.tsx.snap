// Vitest Snapshot v1

exports[`<Button /> > should render in primary colour and medium size by default 1`] = `
<button>
  Hello
</button>
`;
